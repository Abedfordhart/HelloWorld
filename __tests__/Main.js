import Native from 'react-native';
import React from 'react';
import HelloWorld from '../App/index.js';
// import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
// import TestUtils from 'react-addons-test-utils';

describe('HelloWorld', () => {
  describe('<HelloWorld />', () => {
    global.document = {
      getElementById: function() {
        return {
          on: function() {}
      }}
    }
  })

  // let wrapper = shallow(<HelloWorld />);

  it('renders correctly', () => {
    const tree = renderer.create(
      <HelloWorld />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('is a stateful class component.', () => {
    expect(React.Component.isPrototypeOf(HelloWorld)).toBe(true);
  });

  test('one plus one equals two', () => {
    expect(1 + 1).toBe(2)
  });

  // test('contains a Provider to send global state down to downstream components.', () => {
  //   const parentAdmin = TestUtils.renderIntoDocument(<HelloWorld />);
  //   expect(wrapper.contains(<provider></provider>)).toEqual(true)
  // });
})