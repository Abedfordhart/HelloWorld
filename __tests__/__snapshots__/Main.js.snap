// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "column-reverse",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 10,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5FCFF",
              "bottom": 0,
              "flex": 1,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 25,
              }
            }
          >
            Where are you headed?
          </Text>
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 1,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 15,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              When do you need a guide?
            </Text>
          </View>
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 1,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 15,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Date
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#bdc6cf",
                  "borderBottomWidth": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                },
                undefined,
              ]
            }
          >
            <TextInput
              id="date"
              onChangeText={[Function]}
              placeholder="YYYY-MM-DD"
              selectionColor="#86939e"
              style={
                Array [
                  Object {
                    "color": "#86939e",
                    "fontSize": 17.5,
                    "height": 36,
                    "width": 750,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 1,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 15,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Hours
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#bdc6cf",
                  "borderBottomWidth": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                },
                undefined,
              ]
            }
          >
            <TextInput
              id="hours"
              onChangeText={[Function]}
              placeholder="9AM-5PM"
              selectionColor="#86939e"
              style={
                Array [
                  Object {
                    "color": "#86939e",
                    "fontSize": 17.5,
                    "height": 36,
                    "width": 750,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 1,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 15,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Where?
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#bdc6cf",
                  "borderBottomWidth": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                },
                undefined,
              ]
            }
          >
            <TextInput
              id="where"
              onChangeText={[Function]}
              placeholder="Where do you want to go?"
              selectionColor="#86939e"
              style={
                Array [
                  Object {
                    "color": "#86939e",
                    "fontSize": 17.5,
                    "height": 36,
                    "width": 750,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {},
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 1,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 15,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              How many travelers?
            </Text>
          </View>
          <View
            style={
              Object {
                "width": 350,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "1",
                    "textColor": undefined,
                    "value": 1,
                  },
                  Object {
                    "label": "2",
                    "textColor": undefined,
                    "value": 2,
                  },
                  Object {
                    "label": "3",
                    "textColor": undefined,
                    "value": 3,
                  },
                  Object {
                    "label": "4",
                    "textColor": undefined,
                    "value": 4,
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginLeft": 15,
                    "marginRight": 15,
                  },
                  Object {
                    "shadowColor": "rgba(0,0,0, .4)",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 1,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#9E9E9E",
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "padding": 19,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "backgroundColor": "#FF8C00",
                      },
                      undefined,
                      false,
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 20,
                          },
                          undefined,
                          false,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    EXPLORE
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
